### PR TITLE
feat: Maintain database metadata

### PR DESCRIPTION
**Pull request by JigarJoshi**
Start to manage metadata for the database/project.

### CODE REVIEW

File: server/metadata/tenant.go [hunk 1]
```diff
@@ -611,7 +620,24 @@ func (tenant *Tenant) CreateDatabase(ctx context.Context, tx transaction.Tx, dbN
 
 	// otherwise, proceed to create the database if there are concurrent requests on different workers then one of
 	// them will fail with duplicate entry and only one will succeed.
-	_, err := tenant.metaStore.CreateDatabase(ctx, tx, dbName, tenant.namespace.Id())
+	dbId, err := tenant.metaStore.CreateDatabase(ctx, tx, dbName, tenant.namespace.Id())
+	if config.DefaultConfig.Auth.EnableNamespaceIsolation {
```
Comment (himank) [inline server/metadata/tenant.go:624]: do we need this check here? if accessToken is present then we store it by default?

File: server/metadata/tenant.go [hunk 2]
```diff
@@ -601,7 +610,7 @@ func (tenant *Tenant) GetNamespace() Namespace {
 // metadata version once the commit is successful so reloading happens at the next call when a transaction sees a stale
 // tenant version. This applies to the reloading mechanism on all the servers. It returns "true" If the database already
 // exists, else "false" and the error.
-func (tenant *Tenant) CreateDatabase(ctx context.Context, tx transaction.Tx, dbName string) (bool, error) {
+func (tenant *Tenant) CreateDatabase(ctx context.Context, tx transaction.Tx, dbName string, accessToken *types.AccessToken) (bool, error) {
```
Comment (himank) [inline server/metadata/tenant.go:613]: What about passing DatabaseMetadata as a param and then we can have a helper method to build DatabaseMetadata which is called by the runner? This way if we add anything to the database metadata then we don't need to change the function signature.

File: server/metadata/database.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+type DatabaseMetadata struct {
+	Version   uint32
```
Comment (himank) [inline server/metadata/database.go:18]: We don't need to have a version inside it that can go inside the TableData struct.

File: server/metadata/namespaces.go [hunk 4]
```diff
@@ -27,6 +28,7 @@ import (
 
 const (
 	NamespaceSubspaceName = "namespace"
+	DatabaseKey           = "db"
```
Comment (himank) [inline server/metadata/namespaces.go:31]: Can you also add `namespace`, `db` to the reserved/restrictions list.  Also, this key is already defined in `dictionary.go`, better to use that.

File: server/metadata/tenant.go (hunk 1 above)
Comment (JigarJoshi) [reply server/metadata/tenant.go]: if namespace isolation is enabled, access token must be not nil here.

File: server/metadata/tenant.go (hunk 2 above)
Comment (JigarJoshi) [reply server/metadata/tenant.go]: sounds good. made relevant updates.

File: server/metadata/database.go (hunk 3 above)
Comment (JigarJoshi) [reply server/metadata/database.go]: right. deleted it.

File: server/metadata/namespaces.go (hunk 4 above)
Comment (JigarJoshi) [reply server/metadata/namespaces.go]: I made some validation. please let me know if there is any place else I missed it.

File: server/metadata/tenant.go (hunk 1 above)
Comment (JigarJoshi) [reply server/metadata/tenant.go]: this is no longer valid with metadata being created from outside.

**Review by himank — APPROVED** [review]

**Pull request merged**